### PR TITLE
DM-36365: Use astropy caching for downloaded data

### PR DESCRIPTION
**Pull request by mfisherlevine**

### CODE REVIEW

**Comment by jeremyneveu** [comment]
Hi @mfisherlevine . I tested the code, I am ok with the changes. I have just repaired one doctest. If you're done, you can merge.

**Comment by jeremyneveu** [comment]
PS: I am not sure... for the PIP install process I might have to repackage the code, right ?

**Comment by mfisherlevine** [comment]
> PS: I am not sure... for the PIP install process I might have to repackage the code, right ?

As discussed OOB, Jérémy was pointing out that we needed to bump the version number, which I've now done.

File: getCalspec/rebuild.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
            "download_all_data",
            ]
 
-CALSPEC_URL = "https://www.stsci.edu/hst/instrumentation/reference-data-for-calibration-and-tools/astronomical-catalogs/calspec.html"
+CALSPEC_URL = "https://www.stsci.edu/hst/instrumentation/reference-data-for-calibration-and-tools/astronomical-catalogs/calspec.html"  # noqa: E501
```
Comment (ktlim) [inline getCalspec/rebuild.py]:
```suggestion
CALSPEC_URL = (
    "https://www.stsci.edu/hst/instrumentation/"
    "reference-data-for-calibration-and-tools/"
    "astronomical-catalogs/calspec.html"
)
```

File: getCalspec/getCalspec.py [hunk 2]
```diff
@@ -137,32 +138,25 @@ def _santiseName(self, name):
...
-
         spectrum_file_name = self._santiseName(self.Name) + self.STIS.replace('*', '') + ".fits"
-        output_file_name = os.path.join(output_directory, spectrum_file_name)
-        if not os.path.isfile(output_file_name):
-            url = CALSPEC_ARCHIVE+spectrum_file_name
-            try:
-                request.urlretrieve(url, output_file_name)
-            except HTTPError as e:
-                raise RuntimeError(f"Failed to get data for {self.Name} from {url}") from e
+        url = CALSPEC_ARCHIVE+spectrum_file_name
```
Comment (ktlim) [inline getCalspec/getCalspec.py]:
```suggestion
        url = CALSPEC_ARCHIVE + spectrum_file_name
```
Spaces around (non-multiplicative) binary operator.

**Review by ktlim — APPROVED** [review]

**Pull request merged**